DOC: Refinement of confounds documentation after #1877 (#1906)
poldracklab/fmriprep: docs/outputs.rst

Spike regressors are stored in separate ``motion_outlier_XX`` columns.

**Discrete cosine-basis regressors**.
Physiological and instrumental (scanner) noise sources are generally present in fMRI
data, typically taking the form of low-frequency signal drifts.
To account for these drifts, temporal high-pass filtering is the immediate option.
Alternatively, low-frequency regressors can be included in the statistical model to account
for these confounding signals.
Using the :abbr:`DCT (discrete cosine transform)` basis functions, *fMRIPrep* generates
these low-frequency predictors:

- ``cosine_XX`` - DCT-basis regressors.

One characteristic of the cosine regressors is that they are identical for two different
datasets with the same :abbr:`TR (repetition time)` and the same *effective* number of
time points (*effective* length).
It is relevant to mention *effective* because initial time points identified as *nonsteady
states* are removed before generating the cosine regressors.

.. caution::
    If the analysis includes your own high-pass filtering step, the ``cosine_XX``
    regressors may interfere (meaning, they may re-introduce drifts) if included

>     in the design matrix.

Reviewer: I'm a little skeptical of this claim of reintroducing drifts.

What about:

"If your analysis includes separate high-pass filtering, do not also include ``cosine_XX`` regressors in your design matrix."

.. admonition:: See also

    - A detailed explanation about temporal high-pass filtering is provided with
      the `BrainVoyager User Guide

>       <http://www.brainvoyager.com/bvqx/doc/UsersGuide/Preprocessing/TemporalHighPassFiltering.html>`_.

Reviewer: ```suggestion
      <https://www.brainvoyager.com/bvqx/doc/UsersGuide/Preprocessing/TemporalHighPassFiltering.html>`_.
```

    Entries that are not saved in the data file for denoising are still stored in metadata with the
    ``dropped`` prefix.

.. caution::
    Only a subset of these CompCor decompositions should be used for further denoising.
    The original Behzadi aCompCor implementation [Behzadi2007]_ can be applied using
    components from the combined masks, while the more recent Muschelli implementation
    [Muschelli2014]_ can be applied using
    the :abbr:`WM (white matter)` and :abbr:`CSF (cerebro-spinal fluid)` masks.

>     To determine the provenance of each component, consult the metadata file (described above).

Reviewer 2: It might be worth mentioning the option to include either a fixed number of regressors or sufficient regressors to explain some minimum fraction of variance (e.g., 50%), both of which are possible using the confounds generated by fMRIPrep.

For instance:
```
.. caution::
    There are many valid ways of selecting CompCor components for further denoising.
    In general, the components with the largest singular values (i.e., those that
    explain the largest fraction of variance in the data) should be selected.
    *fMRIPrep* outputs components in descending order of singular value.
    Common approaches include selecting a fixed number of components (e.g., the
    first 5 or 6), using a quantitative or qualitative criterion (e.g., elbow, broken
    stick, or condition number), or using sufficiently many components that a minimum
    cumulative fraction of variance is explained (e.g., 50%)
```

.. caution::
    Similarly, if you are using anatomical or temporal CompCor it may not make sense
    to use global signal regressors (``csf``, ``white_matter``, or ``global_signal``) -

>     see `#1049 <https://github.com/poldracklab/fmriprep/issues/1049>`_.

Reviewer 2: Although I'd probably not recommend including both compartmental aCompCor time series and mean compartment (CSF, WM) time series in a single model, I don't agree that it doesn't make sense to use the overall global signal. In particular, @lindenmp and colleagues found in their evaluation that there can be an added benefit of GSR+aCompCor over and above aCompCor alone.

See the green aCompCor bars, particularly in B and C (fig. 1 from Parkes et al.):
<img src="https://user-images.githubusercontent.com/13735271/70570639-2416db00-1b51-11ea-8729-191903c07146.png" width="400" />

Author: /cc @satra

    see `#1049 <https://github.com/poldracklab/fmriprep/issues/1049>`_.
    Conversely, using the overall ``global_signal`` confound in addition to CompCor's
    regressors can be beneficial (see [Parkes2018]_).

.. danger::
    *fMRIPrep* does high-pass filtering before running anatomical or temporal CompCor.
    Therefore, when using CompCor regressors, the corresponding ``cosine_XX`` regressors
    should also be included in the design matrix.

.. admonition:: See also

    This didactic `discussion on NeuroStars.org
    <https://neurostars.org/t/fmrirep-outputs-very-high-number-of-acompcors-up-to-1000/5451>`__
    where Patrick Sadil gets into details about PCA and how that base technique applies
    to CompCor in general and *fMRIPrep*'s implementation in particular.

**AROMA confounds**.
:abbr:`AROMA (Automatic Removal Of Motion Artifacts)` is an :abbr:`ICA (independent components analysis)`
based procedure to identify confounding time series related to head-motion [Prium2015]_.
ICA-AROMA can be enabled with the flag ``--use-aroma``.

- ``aroma_motion_XX`` - the motion-related components identified by ICA-AROMA.

> .. danger::

Reviewer 3: Time to bring up the debate again.
If we abide by [Hallquist](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3759585/pdf/nihms-492938.pdf), then we should mention that we should recalculate the confounds listed in [#1905](https://github.com/poldracklab/fmriprep/issues/1905#issuecomment-562906047) following the strategies listed on [neurostars](https://neurostars.org/t/fmriprep-ica-aroma-filtering-including-wm-csf-etc-confounds-in-fsl-regfilt/3137/3?u=jdkent). However, if you trust the [simulations I ran](https://github.com/poldracklab/fmriprep-notebooks/blob/9933a628dfb759dc73e61701c144d67898b92de0/05%20-%20Discussion%20AROMA%20confounds%20-%20issue-817%20%5BJ.%20Kent%5D.ipynb), then using the original components would be fine. This problem will eventually need to be settled, but I think presenting both sides currently is okay.

Reviewer 3: In additional Hallquist, This other paper by [Linquist](https://onlinelibrary.wiley.com/doi/full/10.1002/hbm.24528) is important to add. They show the detrimental impact of using "modular pipelines", which in this case means denoising in multiple steps.

    see `#1049 <https://github.com/poldracklab/fmriprep/issues/1049>`_.
    Conversely, using the overall ``global_signal`` confound in addition to CompCor's
    regressors can be beneficial (see [Parkes2018]_).

.. danger::
    *fMRIPrep* does high-pass filtering before running anatomical or temporal CompCor.
    Therefore, when using CompCor regressors, the corresponding ``cosine_XX`` regressors
    should also be included in the design matrix.

.. admonition:: See also

    This didactic `discussion on NeuroStars.org
    <https://neurostars.org/t/fmrirep-outputs-very-high-number-of-acompcors-up-to-1000/5451>`__
    where Patrick Sadil gets into details about PCA and how that base technique applies
    to CompCor in general and *fMRIPrep*'s implementation in particular.

**AROMA confounds**.
:abbr:`AROMA (Automatic Removal Of Motion Artifacts)` is an :abbr:`ICA (independent components analysis)`
based procedure to identify confounding time series related to head-motion [Prium2015]_.
ICA-AROMA can be enabled with the flag ``--use-aroma``.

- ``aroma_motion_XX`` - the motion-related components identified by ICA-AROMA.

.. danger::

>     If you are already using ICA-AROMA's cleaned-data (``~desc-smoothAROMAnonaggr_bold.nii.gz``),

Reviewer: "ICA-AROMA's cleaned-data" is weird. Both the `'s` and the hyphen between cleaned and data. You could do

```suggestion
    If you are already using ICA-AROMA-cleaned data (``~desc-smoothAROMAnonaggr_bold.nii.gz``),
```

Or perhaps

```suggestion
    If you are already using AROMA-cleaned data (``~desc-smoothAROMAnonaggr_bold.nii.gz``),
```

    do not include ICA-AROMA confounds during your design specification or denoising procedure.

>     Additionally, as per [Hallquist2013]_, when using ICA-AROMA's cleaned-data most of the

Reviewer 3: ```suggestion
    Additionally, as per [Hallquist2013]_ and [Lindquist2019]_, when using ICA-AROMA's cleaned-data most of the
```

>     Additionally, as per [Hallquist2013]_, when using ICA-AROMA's cleaned-data most of the

Author: ```suggestion
    Additionally, as per [Hallquist2013]_ and [Lindquist2019]_, when using AROMA-cleaned data most of the
```

    Entries that are not saved in the data file for denoising are still stored in metadata with the
    ``dropped`` prefix.

.. caution::
    Only a subset of these CompCor decompositions should be used for further denoising.
    The original Behzadi aCompCor implementation [Behzadi2007]_ can be applied using
    components from the combined masks, while the more recent Muschelli implementation
    [Muschelli2014]_ can be applied using
    the :abbr:`WM (white matter)` and :abbr:`CSF (cerebro-spinal fluid)` masks.
    To determine the provenance of each component, consult the metadata file (described above).

.. caution::
    Similarly, if you are using anatomical or temporal CompCor it may not make sense
    to use global signal regressors (``csf``, ``white_matter``, or ``global_signal``) -
    see `#1049 <https://github.com/poldracklab/fmriprep/issues/1049>`_.

.. danger::
    *fMRIPrep* does high-pass filtering before running anatomical or temporal CompCor.
    Therefore, when using CompCor regressors, the corresponding ``cosine_XX`` regressors

>     should be included too in the design matrix.

Reviewer: ```suggestion
    should also be included in the design matrix.
```

.. admonition:: See also

    This didactic `discussion on NeuroStars.org
    <https://neurostars.org/t/fmrirep-outputs-very-high-number-of-acompcors-up-to-1000/5451>`__
    where Patrick Sadil gets into details about PCA and how that base technique applies
    to CompCor in general and *fMRIPrep*'s implementation in particular.

**AROMA confounds**.
:abbr:`AROMA (Automatic Removal Of Motion Artifacts)` is an :abbr:`ICA (independent components analysis)`
based procedure to identify confounding time series related to head-motion [Prium2015]_.
ICA-AROMA can be enabled with the flag ``--use-aroma``.

- ``aroma_motion_XX`` - the motion-related components identified by ICA-AROMA.

.. danger::
    If you are already using ICA-AROMA cleaned data (``~desc-smoothAROMAnonaggr_bold.nii.gz``),

>     do not include ICA-AROMA confounds during your design specification or denoising procedure.

Reviewer: Let's mention that dummy scans are dropped before AROMA is performed, and therefore any analysis of these must drop the same dummy scans.

Author: Done

     Movement‐Related effects in fMRI time‐series. Magnetic Resonance in Medicine. 1996.
     doi:`10.1002/mrm.191035031 <https://doi.org/10.1002/mrm.1910350312>`_

  .. [Hallquist2013] Hallquist MN, Hwang K, Luna B. The Nuisance of Nuisance Regression.
     NeuroImage. 2013. doi:`10.1016/j.neuroimage.2013.05.116
     <https://doi.org/10.1016/j.neuroimage.2013.05.116>`_

> 

Reviewer 3: ```suggestion

.. [Lindquist2019] Lindquist, MA, Geuter, S, and Wager, TD, Caffo, BS,
Modular preprocessing pipelines can reintroduce artifacts into fMRI data.
Human Brain Mapping. 2019. doi: `10.1002/hbm.24528
<https://doi.org/10.1002/hbm.24528>`_

```
